tests/agent_server/deploy/e2e/test_autostarted.py

```diff
@@ -244,3 +253,750 @@ def repaired() -> bool:
+
+    # Retrieve the current processes, we should have more processes than `start_children`
+    children_after_deployment = get_process_state(current_pid)
+    assert len(children_after_deployment) == 1
+    assert len(children_after_deployment.values()) == 1
+    current_children_after_deployment = filter_relevant_processes(children_after_deployment)
+
+    result = await client.agent_action(tid=environment, name="agent1", action=AgentAction.pause.value)
+    assert result.code == 200
+
+    # Let's pretend that the server crashes
+    await asyncio.wait_for(server.stop(), timeout=20)
+    ibl = InmantaBootloader(configure_logging=False)
+    async_finalizer.add(partial(ibl.stop, timeout=20))
+    # Let's restart the server
+    await ibl.start()
+
+    async def wait_for_scheduler() -> bool:
+        current_children_mapping = get_process_state(current_pid)
+        assert len(current_children_mapping) == 1
+        assert len(current_children_mapping.values()) == 1
+        current_children = filter_relevant_processes(current_children_mapping)
+        # Only the Scheduler should be there as no agent will be recreated given that it's paused
+        return len(current_children) == 1
+
+    # Wait for the scheduler
+    await retry_limited(wait_for_scheduler, 5)
+
+    # Everything should be consistent in DB: the agent should still be paused
+    result = await client.list_agents(tid=environment)
+    assert result.code == 200
+    assert len(result.result["agents"]) == 2
+    expected_agents_status = {e["name"]: e["paused"] for e in result.result["agents"]}
+    assert set(expected_agents_status.keys()) == {const.AGENT_SCHEDULER_ID, "agent1"}
+    assert not expected_agents_status[const.AGENT_SCHEDULER_ID]
+    assert expected_agents_status["agent1"]
+
+    # Let's recheck the number of processes after restarting the server
+    children_after_restart = get_process_state(current_pid)
+    assert len(children_after_restart) == 1
+    assert len(children_after_restart.values()) == 1
+    current_children_children_after_restart = filter_relevant_processes(children_after_restart)
+
+    assert (
+        len(current_children_children_after_restart) == len(current_children_after_deployment) - 2
+    ), "These processes should be present: The Scheduler, the fork server and the actual agent!"
+    for children in current_children_children_after_restart.values():
+        assert children.is_running()
+
+    result = await client.resource_list(environment, deploy_summary=True)
+    assert result.code == 200
+    summary = result.result["metadata"]["deploy_summary"]
+    assert summary["total"] == 1, f"Unexpected summary: {summary}"
+    # FIXME this should be fixed -> old resource is still in deploying state, should be available
+    # Uncomment this once fixed, see https://github.com/inmanta/inmanta-core/issues/8216
+    # assert summary["by_state"]["available"] == 1, f"Unexpected summary: {summary}"
```

Comment:
I would assert that it is in deploying, as this doesn't really test anything like this. It could be deploying for all we care